go/common/l1_types.go

```diff
@@ -43,165 +37,73 @@ type L1Tx struct {
 	Dest   common.Address
 }
 
-type (
-	EncodedL1Tx  []byte
-	Transactions []*L1Tx
-)
+type L1Tx = types.Transaction
+
+func NewL1Tx(data L1TxData) *L1Tx {
+	enc, err := rlp.EncodeToBytes(data)
+	if err != nil {
+		panic(err)
+	}
+	return types.NewTx(&types.LegacyTx{
+		Nonce:    RndBtw(0, ^uint64(0)/2),
+		Value:    big.NewInt(1),
+		Gas:      1,
+		GasPrice: big.NewInt(1),
+		Data:     enc,
+	})
+}
 
-type Header struct {
-	ParentHash L1RootHash
-	Miner      NodeID // this is actually coinbase
-	Nonce      Nonce
+func TxData(tx *L1Tx) L1TxData {
+	data := L1TxData{}
+	err := rlp.DecodeBytes(tx.Data(), &data)
+	if err != nil {
+		panic(err)
+	}
+	return data
 }
 
-// todo - replace with the ethereum Block
-type Block struct {
-	Header       *Header
-	Transactions Transactions
+type (
+	EncodedL1Tx  []byte
+	Transactions types.Transactions
+)
 
-	// ReceiveTime time.Time
+type Header = types.Header
```

Comment:
Is this redundant?

Reply:
I'll get rid of these aliases. They served their purpose